processing/src/main/java/io/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -150,31 +148,28 @@ public String toString()
         .build();
   }
 
-  public static enum Ordering implements Comparator<Number>
+  public enum Ordering implements Comparator<Comparable>
   {
     // ensures the following order: numeric > NaN > Infinite
     numericFirst {
```

Comment:
It's too smart, suggested to change it to `NUMERIC_FIRST` constant with anonymous impl, and set it in constructor if the given ordering is `"numericFirst"`.

Also, this ordering is not null-proof, unlike the default comparator. It will throw NPE.